src/http/common.cr

```diff
@@ -95,9 +109,28 @@ module HTTP
       true
     end
   end
+
+  def self.parse_time(time_str : String)
+    patterns = ["%a, %d %b %Y %H:%M:%S %z", "%A, %d-%b-%y %H:%M:%S %z", "%a %b %e %H:%M:%S %Y"]
+    patterns.each do |pattern|
+      begin
+        return Time.parse(time_str, pattern)
+      rescue
+        # TODO: do something here?
+      end
+    end
+
+    return nil
+  end
+
+  def self.rfc1123_string(time : Time) : String
```

Comment:
I've been reading a bit about this in [1] and [2] and i'm not really sure how it works, it just made me a bit more confused. 

The reason i named it 1153 is this part in [1]: "though they MUST only generate the RFC 1123 format for representing HTTP-date values in header fields"

[1] http://tools.ietf.org/html/rfc2616#section-3.3.1
[2] https://tools.ietf.org/html/rfc7231#section-7.1.1.1